swarm/api/http/server_test.go

```diff
@@ -18,21 +18,78 @@ package http_test
 
 import (
 	"bytes"
+	"crypto/rand"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
 	"strings"
 	"sync"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/swarm/api"
 	swarm "github.com/ethereum/go-ethereum/swarm/api/client"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 	"github.com/ethereum/go-ethereum/swarm/testutil"
 )
 
+func init() {
+	log.Root().SetHandler(log.CallerFileHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true)))))
+}
+
+func TestBzzGetDb(t *testing.T) {
+	srv := testutil.NewTestSwarmServer(t)
+	defer srv.Close()
+
+	keybytes := make([]byte, common.HashLength) // nearest we get to source of info
+	_, err := rand.Read(keybytes)
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
+		return
+	}
+
+	url := fmt.Sprintf("%s/bzz-db:/%s/42", srv.URL, fmt.Sprintf("%x", keybytes))
+	resp, err := http.Post(url, "application/octet-stream", nil)
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
+		return
+	}
+	b, err := ioutil.ReadAll(resp.Body)
+	fmt.Printf("Create: %s : %s\n", resp.Status, b)
+
+	url = fmt.Sprintf("%s/bzz-db:/%s", srv.URL, fmt.Sprintf("%x", keybytes))
+	data := []byte("foo")
+	resp, err = http.Post(url, "application/octet-stream", bytes.NewReader(data))
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
+		return
+	}
+	b, err = ioutil.ReadAll(resp.Body)
+	fmt.Printf("Update: %s : %s\n", resp.Status, b)
+
+	url = fmt.Sprintf("%s/bzz-db-raw:/%s", srv.URL, fmt.Sprintf("%x", keybytes))
+	resp, err = http.Get(url)
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
+		return
+	}
+	b, err = ioutil.ReadAll(resp.Body)
```

Comment:
Should there not be an assertion checking the response body?

Reply:
yeah, but will make it later when the general setup of the api is approved. "In progress"...